java/client/src/main/java/glide/api/commands/Transaction.java

```diff
@@ -0,0 +1,149 @@
+package glide.api.commands;
+
+import glide.api.models.commands.InfoOptions;
+import glide.api.models.commands.SetOptions;
+import glide.managers.models.Command;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Singular;
+
+/**
+ * Base class encompassing shared commands for both standalone and cluster mode implementations in a
+ * transaction. Transactions allow the execution of a group of commands in a single step.
+ *
+ * <p>Command Response: An array of command responses is returned by the client exec command, in the
+ * order they were given. Each element in the array represents a command given to the transaction.
+ * The response for each command depends on the executed Redis command. Specific response types are
+ * documented alongside each method.
+ *
+ * @example transaction = new Transaction.Builder() .set("key", "value"); .get("key"); .build();
+ *     Object[] result = client.exec(transaction).get(); assertEqual(new Object[] {OK , "value"});
+ */
+@Builder
+@Getter
+@EqualsAndHashCode
+public class Transaction {
+    @Singular List<Command> commands = new LinkedList<>();
+
+    public static class TransactionBuilder {
+
+        /**
+         * Executes a single command, without checking inputs. Every part of the command, including
+         * subcommands, should be added as a separate value in args.
+         *
+         * @remarks This function should only be used for single-response commands. Commands that don't
+         *     return response (such as SUBSCRIBE), or that return potentially more than a single
+         *     response (such as XREAD), or that change the client's behavior (such as entering pub/sub
+         *     mode on RESP2 connections) shouldn't be called using this function.
+         * @example Returns a list of all pub/sub clients:
+         *     <pre>
+         * Object result = client.customCommand(new String[]{"CLIENT","LIST","TYPE", "PUBSUB"}).get();
+         * </pre>
+         *
+         * @param args arguments for the custom command
+         * @return a CompletableFuture with response result from Redis
+         */
+        public TransactionBuilder customCommand(String[] args) {
+            createOrAddCommand(Command.customCommand(args));
+            return this;
+        }
+
+        /**
+         * Ping the Redis server.
+         *
+         * @see <a href="https://redis.io/commands/ping/">redis.io</a> for details.
+         * @return the String "PONG"
+         */
+        public TransactionBuilder ping() {
+            createOrAddCommand(Command.ping());
+            return this;
+        }
+
+        /**
+         * Ping the Redis server.
+         *
+         * @see <a href="https://redis.io/commands/ping/">redis.io</a> for details.
+         * @param msg - the ping argument that will be returned.
```

Comment:
Ensure that you inherit all javadoc fixes from another PRs

Reply:
I will move all javadocs to interfaces too.